autodoc/data/admin-api.lua

```diff
@@ -390,6 +391,33 @@ return {
         },
       },
     },
+    metadata = {
+      title = [[ Metadata ]],
+      description = [[ Developer-supplied data. ]],
+      ["/metadata/plugins/:name"] = {
```

Comment:
I was discussing this with @darrenjennings , who makes the good point that while this is RESTful and consistent with the rest of the Admin API, it also requires the frontend to make upwards of 40 API calls just to render a list of plugins like https://manager-steady.kong-cloud.com/default/plugins/select/. That's not really practical, but the alternatives don't fit well into the Admin API as it exists today.

Reply:
a few alternatives off the top of my head:
- gather the metadata of _all_ installed plugins on a single call (just `/metadata/plugins/`, without a name?)
- let the client ask for specific metadata fields (`/metadata/plugins/<name>/?f[]=icon&f[]=author&f[]=description&f[]=licence`. optional plugin name, as on previous point).  after all, only known fields can be rendered on the UI, right?
- use a registry.  not only can be managed, but could be GraphQL if you wish!

personally, I still think a registry is the best solution, but i'm aware that the decision hinges mostly on what's the desired process for external plugin writers.